app/controllers/reason-for-appeal.ts

```diff
@@ -0,0 +1,67 @@
+
+import { NextFunction, Request, Response, Router } from 'express';
+import { paths } from '../paths';
+import { Events } from '../service/ccd-service';
+import UpdateAppealService from '../service/update-appeal-service';
+import { homeOfficeDecisionValidation } from '../utils/fields-validations';
+import { daysToWaitUntilContact } from './confirmation-page';
+
+function getReasonForAppeal(req: Request, res: Response, next: NextFunction) {
+  try {
+    return res.render('case-building/reasons-for-appeal/decision-appeal.njk');
+  } catch (e) {
+    next(e);
+  }
+}
+
+function postReasonForAppeal(updateAppealService: UpdateAppealService) {
+  return async function(req: Request, res: Response, next: NextFunction) {
+    try {
+      const validation = homeOfficeDecisionValidation(req.body);
+      if (validation != null) {
+        return res.render('case-building/reasons-for-appeal/decision-appeal.njk', {
+          errorList: Object.values(validation),
+          error: validation
+
+        });
+      }
+      req.session.appeal.caseBuilding = {
+        decision: req.body.moreDetail
+      };
+      // TODO Save to CCD.
+      await updateAppealService.submitEvent(Events.UPLOAD_RESPONDENT_EVIDENCE, req);
+      return res.redirect('case-building/reasons-for-appeal/check-and-send.njk');
+    } catch (e) {
+      next(e);
+    }
+  };
+}
```

Comment:
Is this story just for the confirmation page? Also, template naming is a bit confusing? Instead of decision-appeal.njk shouldnt it be named to reasons-for-appeal.njk?